plugins/powerpc/powerpc_arith.ml

```diff
@@ -0,0 +1,420 @@
+
+open Powerpc.Std
+
+(** Fixed-Point Arithmetic Instructions - Negate
+    Page 72 of IBM Power ISATM Version 3.0 B
+    example:
+    7c 22 00 d0   neg  r1, r2
+    7c 22 00 d1   neg. r1, r2 *)
+let neg cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  RTL.[ rt := lnot ra + one ]
+
+(** Fixed-Point Arithmetic Instructions - Substract From
+    Page 69 of IBM Power ISATM Version 3.0 B
+    example:
+    7c 22 18 50    subf  r1, r2, r3
+    7c 22 18 51    subf. r1, r2, r3 *)
+let subf cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  let rb = unsigned cpu.reg ops.(2) in
+  RTL.[rt := (lnot ra) + rb + one]
+
+(** Fixed-Point Arithmetic Instructions - Substract From Immediate Carrying
+    Page 70 of IBM Power ISATM Version 3.0 B
+    example:
+    20 22 10 92    subfic r1, r2, 4242 *)
+let subfic cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let ra = unsigned cpu.reg ops.(1) in
+  let si = unsigned imm ops.(2) in
+  RTL.[
+    rt := (lnot ra) + si + one;
```

Comment:
You're inconsistent with using parentheses around the unary operators. They are not needed here, though you're still using them. At the same time, at other places you're not using. People may start to believe that parentheses here matter.